io/private/stac/Item.cpp

```diff
@@ -0,0 +1,547 @@
+        const std::string& driver) const
+    {
+        Options readerOptions;
+        if (readerArgs.contains(driver)) {
+            NL::json args = m_utils.jsonValue(readerArgs, driver);
+            for (auto& arg : args.items())
+            {
+                std::string key = arg.key();
+                NL::json val = arg.value();
+                NL::detail::value_t type = val.type();
+
+                // if value is of type string, dump() returns string with
+                // escaped string inside and kills pdal program args
+                std::string v;
+                if (type == NL::detail::value_t::string)
+                    v = m_utils.jsonValue<std::string>(val);
+                else
+                    v = arg.value().dump();
+                readerOptions.add(key, v);
+            }
+        }
+
+        return readerOptions;
+    }
+
+    std::string Item::extractDriverFromItem(const NL::json& asset) const
+    {
+        std::string output;
+
+        std::map<std::string, std::string> contentTypes =
+        {
+            { "application/vnd.laszip+copc", "readers.copc"}
+        };
+
+        std::string assetPath = m_utils.stacValue<std::string>(
+            asset, "href", m_json);
+        std::string dataUrl = m_utils.handleRelativePath(m_path, assetPath);
+
+        std::string contentType;
+
+        if (asset.contains("type"))
+        {
+            contentType = m_utils.stacValue<std::string>(asset, "type", m_json);
+            for(const auto& ct: contentTypes)
+                if (Utils::iequals(ct.first, contentType))
+                    return ct.second;
+        }
+
+        if (!FileUtils::fileExists(dataUrl))
+        {
+            // Use this to test if dataUrl is a valid endpoint
+            // Try to make a HEAD request and get it from Content-Type
+            try
+            {
+                StringMap headers = m_connector.headRequest(dataUrl);
+                if (headers.find("Content-Type") != headers.end())
+                {
+                    contentType = headers["Content-Type"];
+                    for(const auto& ct: contentTypes)
+                        if (Utils::iequals(ct.first, contentType))
+                            return ct.second;
+                }
+            }
+            catch(std::exception e)
+            {
+                throw stac_error(m_id, "item", "Failed to HEAD " + dataUrl +
+                    ". " + e.what());
+            }
+        }
+
+        // Try to guess from the path
+        std::string driver = m_factory.inferReaderDriver(dataUrl);
+        if (driver.size())
+            return driver;
+
+        return output;
+    }
+
+    void Item::validate()
+    {
+
+        nlohmann::json_schema::json_validator val(
+            [this](const nlohmann::json_uri& json_uri, nlohmann::json& json) {
+                NL::json tempJson = m_connector.getJson(json_uri.url());
+
+                std::lock_guard<std::mutex> lock(m_mutex);
```

Comment:
It doesn't appear that we need `m_mutex` here, or at all as a member variable, of this class.  This is the only place that it's used and it doesn't look like there is any contention possible with this usage.